packages/theme/src/cli/utilities/repl.ts

```diff
@@ -0,0 +1,37 @@
+import {DevServerSession} from './theme-environment/types.js'
+import {presentValue} from './repl/presenter.js'
+import {evaluate} from './repl/evaluater.js'
+import {AbortSilentError} from '@shopify/cli-kit/node/error'
+import {consoleWarn, outputContent, outputDebug, outputInfo, outputToken} from '@shopify/cli-kit/node/output'
+import {renderTextPrompt} from '@shopify/cli-kit/node/ui'
+
+export async function replLoop(themeSession: DevServerSession, themeId: string, url: string) {
+  try {
+    const inputValue = await renderTextPrompt({message: 'Enter a value'})
```

Comment:
May we support the history commands with `renderTextPrompt` (as we may notice in the video below)?

https://github.com/user-attachments/assets/8ae95ee3-e13d-4597-931e-2cd15441c52e

While consistent in the UI components is great, it seems like `readline` provides the exact stateful features we need on the context of a REPL tool ([ref](https://github.com/Shopify/cli/pull/4191#discussion_r1686424514))

What do you think, @jamesmengo? :)